app/questionnaire/router.py

```diff
@@ -258,6 +266,86 @@ def _get_return_to_location_url(
                 "questionnaire.submit_questionnaire", _anchor=return_to_answer_id
             )
 
+    def _get_return_to_for_grand_calculated_summary(
+        self,
+        *,
+        return_to: str | None,
+        return_to_block_id: str | None,
+        location: Location,
+        routing_path: RoutingPath,
+        return_to_answer_id: str | None = None,
+    ) -> str | None:
+        """
+        Builds the return url for a grand calculated summary,
+        and accounts for it possibly being in a different section to the calculated summaries it references
+        """
+        # Type ignore: the section for the grand calculated summary must exist
+        grand_calculated_summary_section = location.section_id
+        if return_to_block_id:
+            grand_calculated_summary_section: str = self._schema.get_section_id_for_block_id(return_to_block_id)  # type: ignore
+        if grand_calculated_summary_section != location.section_id:
+            # the grand calculated summary is in a different section which will have a different routing path
+            routing_path = self._path_finder.routing_path(
+                section_id=grand_calculated_summary_section
+            )
+        if self.can_access_location(
+            # grand calculated summaries don't support repeating sections so no need for list item here
```

Comment:
Is this true?. It is true that a grand summary won't reference calc summary that is in a repeating section unless the grand summary is in a repeating section itself.

Reply:
So I could add list_item_id as the section and list item will be consistent in the case of a repeating section, but as theres a follow up ticket to handle grand calculated summaries with repeating calculated summaries I think it's best I leave as is? I've updated the comment

Reply:
ahhh I see - I misunderstood, I think a new card would be good as in that case there should be a test schema for a grand calculated summary inside a repeat and new sets of tests to cover this scenario

Reply:
cheers Mebin 👍

Reply:
Will be done in https://trello.com/c/gP4302Vl/5733-grand-calculated-summary-support-within-repeating-sections